shenyu-plugin/shenyu-plugin-api/src/main/java/org/apache/shenyu/plugin/api/utils/BodyParamUtils.java

```diff
@@ -72,26 +71,26 @@
             paramNameList.addAll(paramNameMap.keySet());
             paramTypeList.addAll(paramNameMap.values());
         } else {
-            paramTypeList = Arrays.asList(StringUtils.split(parameterTypes, ","));
+            Map<String, Object> paramMap = GsonUtils.getInstance().toObjectMap(body);
+            paramNameList.addAll(paramMap.keySet());
```

Comment:
Repeat code, you should refactor it.

Reply:
There is different map.keyset  to added paramNameList